server/metadata_context_test.go

```diff
@@ -87,6 +92,43 @@ func TestEncodeMetadataNilCheck(t *testing.T) {
 	}
 }
 
+func TestEncodeMetadatagRPCCheck(t *testing.T) {
```

Comment:
I don't think you need to downcase the `g` here; although `gRPC` is the proper initialism in writing, I think you're better off treating it as a word, and writing `TestEncodeMetadataGrpcCheck` here.